### PR TITLE
Add asScrollableContent

### PR DESCRIPTION
**Pull request by M-i-k-e-l**

### CODE REVIEW

File: src/commons/asScrollableContent.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+//@ts-ignore
+import hoistStatics from 'hoist-non-react-statics';
+
+interface ScrollableContentProps {
+  scrollableContentProps?: ScrollViewProps;
+  maximumContentSize?: number;
+}
+
+interface ScrollableContentState {
+  maximumContentSize?: number;
```
Comment (ethanshar) [inline src/commons/asScrollableContent.tsx:13]: Why should the user need to pass the `maximumContentSize`? 
The whole idea is to measure (using onLayout) the content size vs the container size and do the whole work

File: src/commons/asScrollableContent.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+export default function asScrollableContent(WrappedComponent: React.ComponentType<any>) {
+  class ScrollableContent extends Component<PropTypes, ScrollableContentState> {
+    static displayName: string | undefined;
+
+    constructor(props: PropTypes) {
+      super(props);
+      this.state = {
+        maximumContentSize: props.maximumContentSize,
+        contentSize: undefined,
+        scrollEnabled: false
```
Comment (ethanshar) [inline src/commons/asScrollableContent.tsx:29]: we have the scrollEnable state to lock/unlock the scroll that's only internal. 
How would you use this HOC to decide if the to render a gradient at the bottom? or anything at all. 
You don't pass the user back this value in order to make a use of it.

File: src/commons/asScrollableContent.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      };
+    }
+
+    static isScrollEnabled = (contentSize?: number, maximumContentSize?: number) => {
+      return !_.isUndefined(contentSize) && !_.isUndefined(maximumContentSize) && contentSize > maximumContentSize;
+    };
+
+    static getDerivedStateFromProps(nextProps: any, prevState: any) {
```
Comment (ethanshar) [inline src/commons/asScrollableContent.tsx:37]: If you're writing a new component and already has to use `getDerivedStateFromProps` it should already raise a flag. Do we really want this component has to be uncontrolled?

File: src/commons/asScrollableContent.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+interface ScrollableContentState {
+  maximumContentSize?: number;
+  contentSize?: number;
+  scrollEnabled: boolean;
+}
+
+type PropTypes = ScrollableContentProps;
+
+export default function asScrollableContent(WrappedComponent: React.ComponentType<any>) {
+  class ScrollableContent extends Component<PropTypes, ScrollableContentState> {
```
Comment (ethanshar) [inline src/commons/asScrollableContent.tsx:21]: Consider experimenting with functions and hooks

File: src/commons/index.js [hunk 5]
```diff
@@ -16,6 +16,9 @@ module.exports = {
   get UIComponent() {
     return require('./UIComponent').default;
   },
+  get asScrollableContent() {
```
Comment (ethanshar) [inline src/commons/index.js:19]: move it to commons/new
This file will be replaced by commons/new eventually

**Comment by ethanshar** [comment]
@M-i-k-e-l 
Is this PR still relevant?

**Comment by M-i-k-e-l** [comment]
> @M-i-k-e-l
> Is this PR still relevant?

No

**Pull request closed without merging**